### PR TITLE
Migrated to a Cobra-CLI app

### PR DESCRIPTION
**Pull request by gjyotin305**
- Converted to a Cobra app

### CODE REVIEW

File: cli/cli
Comment (SaahilNotSahil) [inline cli/cli:1]: DON'T PUSH BINARY FILES

File: cli/cmd/configure.go [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+// configureCmd represents the configure command
+var configureCmd = &cobra.Command{
+	Use:   "configure",
+	Short: "A brief description of your command",
```
Comment (SaahilNotSahil) [inline cli/cmd/configure.go:18]: Change the help statements

File: cli/cmd/configure.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("---Setup Initiated----")
+		serverFlag, _ := cmd.Flags().GetString("server")
+		keyFlag, _ := cmd.Flags().GetString("key")
```
Comment (SaahilNotSahil) [inline cli/cmd/configure.go:28]: This flag doesn't make sense to me. Please explain

File: cli/cmd/configure.go [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+		emailFlag, _ := cmd.Flags().GetString("email")
+		path, _ := cmd.Flags().GetString("path")
```
Comment (SaahilNotSahil) [inline cli/cmd/configure.go:30]: This flag too

File: cli/cmd/configure.go [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		fmt.Println("Configured SOS server for:", emailArgs)
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(configureCmd)
```
Comment (SaahilNotSahil) [inline cli/cmd/configure.go:159]: Add a new line after this

File: cli/cmd/configure.go [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+	Use:   "configure",
+	Short: "A brief description of your command",
+	Long: `A longer description that spans multiple lines and likely contains examples
+and usage of using your command. For example:
+
+Cobra is a CLI library for Go that empowers applications.
+This application is a tool to generate the needed files
+to quickly create a Cobra application.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("---Setup Initiated----")
```
Comment (SaahilNotSahil) [inline cli/cmd/configure.go:26]: Add a new line after this

File: cli/cmd/login.go [hunk 6]
```diff
@@ -0,0 +1,212 @@
+/*
+Copyright © 2024 NAME HERE <EMAIL ADDRESS>
```
Comment (SaahilNotSahil) [inline cli/cmd/login.go:2]: ??

File: cli/cmd/login.go [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+		if err != nil {
+			fmt.Println(err)
+
+			os.Exit(1)
+		}
+		os.Exit(0)
+	},
+}
+
+func init() {
```
Comment (SaahilNotSahil) [inline cli/cmd/login.go:201]: Push this function to the top of this file below the declaration of loginCmd

File: cli/cmd/login.go [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+				)
+			}
+		}
+	}
+
+	return fmt.Errorf("no default ssh key file free for openpubkey")
+}
+
+// loginCmd represents the login command
+var loginCmd = &cobra.Command{
```
Comment (SaahilNotSahil) [inline cli/cmd/login.go:154]: Push this to the top below the declaration of op

File: cli/cmd/root.go [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+
+
+// rootCmd represents the base command when called without any subcommands
+var rootCmd = &cobra.Command{
+	Use:   "cli",
```
Comment (SaahilNotSahil) [inline cli/cmd/root.go:17]: Change this to "sos"

File: cli/cmd/root.go [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+	Short: "A brief description of your application",
```
Comment (SaahilNotSahil) [inline cli/cmd/root.go:18]: Change the help texts

File: cli/cmd/login.go [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+	Use:   "login",
+	Short: "A brief description of your command",
```
Comment (SaahilNotSahil) [inline cli/cmd/login.go:156]: Change the help texts

File: cli/cmd/root.go [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+to quickly create a Cobra application.`,
+	// Uncomment the following line if your bare application
+	// has an action associated with it:
+	// Run: func(cmd *cobra.Command, args []string) { },
+}
+
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
```
Comment (SaahilNotSahil) [inline cli/cmd/root.go:33]: Use cobra's CheckErr method for error handling

File: cli/cmd/root.go [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+func init() {
+	// Here you will define your flags and configuration settings.
+	// Cobra supports persistent flags, which, if defined here,
+	// will be global for your application.
+
+	// rootCmd.PersistentFlags().StringVar(&cfgFile, "config", "", "config file (default is $HOME/.cli.yaml)")
+
+	// Cobra also supports local flags, which will only run
+	// when this action is called directly.
+	rootCmd.Flags().BoolP("toggle", "t", false, "Help message for toggle")
```
Comment (SaahilNotSahil) [inline cli/cmd/root.go:48]: Explain

**Review by SaahilNotSahil — CHANGES_REQUESTED** [review]
- Get rid of useless comments in the code
- Use cobra's CheckErr method for handling errors

**Comment by gjyotin305** [comment]
@SaahilNotSahil Check now

File: cli/cmd/root.go (hunk 12 above)
Comment (gjyotin305) [reply cli/cmd/root.go:18]: ok done

File: cli/cmd/root.go (hunk 13 above)
Comment (gjyotin305) [reply cli/cmd/root.go]: cobra-cli init produces this by default, overlooked this

File: cli/cmd/login.go (hunk 11 above)
Comment (gjyotin305) [reply cli/cmd/login.go]: done

File: cli/cmd/root.go (hunk 10 above)
Comment (gjyotin305) [reply cli/cmd/root.go]: done

File: cli/cmd/root.go (hunk 9 above)
Comment (gjyotin305) [reply cli/cmd/root.go]: done

**Review by SaahilNotSahil — APPROVED** [review]
LGTM

**Pull request merged**